### PR TITLE
Overhaul cite_easystats()

### PR DESCRIPTION
**Pull request by bwiernik**
1. Provides options to generate citations and bibliographies in plain text, markdown, and biblatex formats.

2. Cites journal articles and packages themselves, including automatically filling in installed package versions.

3. Removes the `as.data.frame()` method. It's not clear to me what the purpose of this was? 

Closes https://github.com/easystats/report/issues/236

### CODE REVIEW

**Comment by bwiernik** [comment]
The failing check doesn't seem to be related to the PR changes? @IndrajeetPatil any idea?

**Review by DominiqueMakowski — APPROVED** [review]

**Comment by IndrajeetPatil** [comment]
Thanks, Brenton. This looks amazing!

I will have a better look after work today. 

And, yeah, the macOS-devel failure is spurious. We should be skipping that particular check.

**Comment by IndrajeetPatil** [comment]
No worries.

I don't understand what's going on here:

```r
.support_unicode <- l10n_info()$`UTF-8` |
  isTRUE(.Options$cli.unicode) |
  nzchar(Sys.getenv("RSTUDIO_USER_IDENTITY"))
```

`.Options$cli.unicode` returns `NULL` on my system, irrespective of whether I load `{cli}` package or not.

Also, what does count of characters in RStudio username (`nzchar(Sys.getenv("RSTUDIO_USER_IDENTITY")`) has to do with UNICODE support? Is RStudio always using Unicode? What if a user (for whatever reason) has changed the global settings to use (e.g.) ASCII?

Screenshot 2022-03-29 at 14 26 33

Just trying to understand the code. These are most likely to be edge cases that users unlikely to run into, so maybe not a big deal at all.

File: inst/easystats_bib.bib [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  volume = {5},
+  number = {56},
+  pages = {2815},
+  issn = {2475-9066},
+  doi = {10.21105/joss.02815},
+  url = {https://joss.theoj.org/papers/10.21105/joss.02815},
+  urldate = {2021-06-30}
+}
+
+@software{effectsizePackage,
```
Comment (IndrajeetPatil) [inline inst/easystats_bib.bib:95]: I think this is incorrect. 

AFAIK, package citations include only authors, and not contributors, so it should not include IP and BW. 

For example, _broom_ has a ton of contributors, but only the three authors are included in the citation.

``` r
citation("broom")
#> 
#> To cite package 'broom' in publications use:
#> 
#>   David Robinson, Alex Hayes and Simon Couch (2022). broom: Convert
#>   Statistical Objects into Tidy Tibbles. R package version 0.7.12.
#>   https://CRAN.R-project.org/package=broom
#> 
#> A BibTeX entry for LaTeX users is
#> 
#>   @Manual{,
#>     title = {broom: Convert Statistical Objects into Tidy Tibbles},
#>     author = {David Robinson and Alex Hayes and Simon Couch},
#>     year = {2022},
#>     note = {R package version 0.7.12},
#>     url = {https://CRAN.R-project.org/package=broom},
#>   }
```

<sup>Created on 2022-03-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Comment (bwiernik) [reply inst/easystats_bib.bib:95]: https://github.com/easystats/effectsize/pull/433 seems appropriate?

**Comment by bwiernik** [comment]
```r
.support_unicode <- l10n_info()$`UTF-8` |
  isTRUE(.Options$cli.unicode) |
  nzchar(Sys.getenv("RSTUDIO_USER_IDENTITY"))
```

There are 3 scenarios where the R console supports Unicode:

1. The R locale is set to UTF-8
2. They have explicitly directed *cli* to use UTF-8 characters via the `option()`
3. The session is running in RStudio, which always supports display of UTF-8 output, even on Windows in non-UTF-8 locales

The options screen you showed Indra is for the encoding to use when saving files, not displaying console output.

This little option is minor, just used to determine whether to display some Unicode characters such as  😀 instead of `:-)`

**Comment by IndrajeetPatil** [comment]
Thanks for the clarifications! Cool stuff.

File: inst/easystats_bib.bib (hunk 1 above)
Comment (strengejacke) [reply inst/easystats_bib.bib:95]: Yeah, that's ok.

Comment (mattansb) [reply inst/easystats_bib.bib:95]: Yup.
Should we go ahead and have the 5 of us as aut on all packages? Is that already the case?

Comment (bwiernik) [reply inst/easystats_bib.bib:95]: I thought it was the case, so I think effectsize might be the only standout, but we should double check

**Comment by bwiernik** [comment]
I generated them with Zotero+BettetBibTeX. It was easier to get all of the formatting correct slightly manually into these two data files rather than trying to do something purely in R

**Comment by bwiernik** [comment]
https://github.com/easystats/report/pull/240

**Comment by bwiernik** [comment]
Okay those are all fixed in #240

**Comment by DominiqueMakowski** [comment]
Looks good, is there an option for bibtex output? 

My only worry is that it will create endless different versions in google scholar that I will have to manually merge 😁 but I guess there are worst problems ^^

**Comment by bwiernik** [comment]
I could add a BibTeX output, but most TeX users use BibLaTeX nowadays, which is dramatically better for software cites. 

For R users, 99% of them are using markdown/pandoc which _really_ should use CSL and not BibTeX for correct citations. That's why I've emphasized that in the docs

**Comment by DominiqueMakowski** [comment]
Guess I need a tutorial on how to use CSL to quit the 1% 🥲

**Comment by bwiernik** [comment]
You're already using CSL to format your citations--that's what pandoc does by default. 

The only change you need to make is to use .YAML or .json for your bibliography file instead of .bib. pandoc internally converts .bib to CSL JSON, but this is a lossy conversion and often yields incorrect results, especially for software citations.

**Pull request merged**